tests/console/yast2_bootloader.pm

```diff
@@ -34,17 +34,7 @@ sub run {
     # YaST2 prompts user to install missing packages found during storage probing.
     # Otherwise YaST2 shows bootloader settings options
     assert_screen([qw(test-yast2_bootloader-1 fs-probing-failed)], 300);
-
-    if (match_has_tag('fs-probing-failed')) {
-        send_key 'alt-d';
-
-        my $keymapping = {'probing-error' => 'ret', 'fs-probing-failed' => 'alt-i'};
-        foreach (qw(probing-error fs-probing-failed)) {
-            assert_screen $_;
-            send_key $keymapping->{$_};
-        }
-        assert_screen 'test-yast2_bootloader-1';
-    }
+    match_has_tag('fs-probing-failed') && $self->ncurses_filesystem_probing('test-yast2_bootloader-1');
```

Comment:
If `ncurses_filesystem_probing` has to be passed the needle already, the `assert_screen` + `match_has_tag` could be moved into `ncurses_filesystem_probing` as well.

Reply:
In such a case we would need to call the `ncurses_filesystem_probing` every single time, including in YaST2 versions missing this feature. I was trying to avoid that. 
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/console/yast2_bootloader.pm#L36

* https://openqa.opensuse.org/tests/1171181#step/yast2_bootloader/4 vs
* http://eris.suse.cz/tests/4829#step/yast2_bootloader/4

Reply:
> http://eris.suse.cz/tests/4850#step/yast2_bootloader/4
> 
> I haven't tested the bootloader update properly, let me find non-JeOS TW job.

* http://eris.suse.cz/tests/4852

Reply:
All it does is move two lines into a function, I don't see a problem with that.

Reply:
Technically that is absolutely correct. I just felt that people who will review the test in the future will directly imply that the probing should happen each run. But that depends on the YaST2 version. I was aiming to highlight the different workflows between YaST2 versions. 
Ofc, it might not help anyway and your suggestion makes perfect sense. :) 
Let me update the code

Reply:
Currently there isn't any explicit check for the YaST version either, only implicit by the use of `'fs-probing-failed'` which older YaST versions don't show. If that's the goal, you could use `$self->ncurses_filesystem_probing('test-yast2_bootloader-1') if is_tumbleweed;` or something like that, with an `assert_screen` for the alternative case.

Reply:
IMHO the implicit one is good enough, which is basically also the same as your first comment suggests. Just the evaluation is directly in `ncurses_filesystem_probing`.

If we would like to go for `is_{distro}`, we would need to touch this exact line in a few days again. This does not seem to be effective. :)

Reply:
updated